web_src/js/svg.js

```diff
@@ -38,18 +40,33 @@ export const svgs = {
 
 const parser = new DOMParser();
 const serializer = new XMLSerializer();
+const parsedSvgs = new Map();
 
-// retrieve a HTML string for given SVG icon name, size and additional classes
-export function svg(name, size = 16, className = '') {
+function getParsedSvg(name) {
+  if (parsedSvgs.has(name)) return parsedSvgs.get(name);
+  const root = parser.parseFromString(svgs[name], 'text/html');
+  const svgNode = root.querySelector('svg');
+  parsedSvgs.set(name, svgNode);
+  return svgNode;
+}
+
+function applyAttributes(node, size, className) {
+  if (size !== 16) node.setAttribute('width', String(size));
+  if (size !== 16) node.setAttribute('height', String(size));
+  if (className) node.classList.add(...className.split(/\s+/));
+  return node;
+}
+
+// returns a SVG node for given SVG icon name, size and additional classes
+export function svgNode(name, size = 16, className = '') {
```

Comment:
There are many magic numbers `16` in code. Maybe we can use a global const. And here `svgNode(name, size = null, className = '')` , then set size to 16 if `size===null`

Reply:
SVG default size of 16 is defined in many places, backend, frontend, svg generator. I think it'd be hard to share this value across those places.

Reply:
This is obsolete, I went back to the `.innerHTML` method